doc: update readline asyncIterator docs (#28425)
nodejs/node: doc/api/readline.md

    description: Symbol.asyncIterator support is no longer experimental.
-->

* Returns: {AsyncIterator}

Create an `AsyncIterator` object that iterates through each line in the input
stream as a string. This method allows asynchronous iteration of

> `readline.Interface` objects through `for await...of` loops.

Reviewer: There are twomore `` `for`-`await`-`of` `` at the end of the doc, in "Example: Read File Stream Line-by-Line" section.

    description: Symbol.asyncIterator support is no longer experimental.
-->

* Returns: {AsyncIterator}

Create an `AsyncIterator` object that iterates through each line in the input
stream as a string. This method allows asynchronous iteration of
`readline.Interface` objects through `for await...of` loops.

Errors in the input stream are not forwarded.

If the loop is terminated with `break`, `throw`, or `return`,
[`rl.close()`][] will be called. In other words, iterating over a
`readline.Interface` will always consume the input stream fully.

> A caveat with using this API is that the performance is currently not on par

Reviewer 2: ```suggestion
Performance is not on par
```

> with the traditional `'line'` event API, and thus it is not recommended for

Reviewer 2: ```suggestion
with the traditional `'line'` event API. Use `line` instead for
```

performance-sensitive applications. This situation is expected to improve in the

> future.

Reviewer 2: I'd remove the last sentence. "This will be better at some undisclosed point in the future, maybe" is better left unsaid IMO. When the situation improves, docs can be updated. (In general, we should avoid relative-time statements like "currently" and "in the future".)
